cmd/newcoin/newcoin.go

```diff
@@ -0,0 +1,220 @@
+func init() {
+	app.Name = "newcoin"
+	app.Usage = "newcoin is a helper tool for creating new fiber coins"
+	app.Version = Version
+	commands := cli.Commands{
+		createCoinCommand(),
+	}
+
+	app.Commands = commands
+	app.EnableBashCompletion = true
+	app.OnUsageError = func(context *cli.Context, err error, isSubcommand bool) error {
+		fmt.Fprintf(context.App.Writer, "error: %v\n\n", err)
+		cli.ShowAppHelp(context)
+		return nil
+	}
+	app.CommandNotFound = func(context *cli.Context, command string) {
+		tmp := fmt.Sprintf("{{.HelpName}}: '%s' is not a {{.HelpName}} "+
+			"command. See '{{.HelpName}} --help'. \n", command)
+		cli.HelpPrinter(app.Writer, tmp, app)
+	}
+}
+
+func createCoinCommand() cli.Command {
+	name := "createcoin"
+	return cli.Command{
+		Name:  name,
+		Usage: "Create a new coin from a template file",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "coin",
+				Usage: "name of the coin to create",
+				Value: "skycoin",
+			},
+			cli.StringFlag{
+				Name:  "template-dir, td",
+				Usage: "template directory path",
+				Value: "./template",
+			},
+			cli.StringFlag{
+				Name:  "coin-template-file, ct",
+				Usage: "coin template file",
+				Value: "coin.template",
+			},
+			cli.StringFlag{
+				Name:  "visor-template-file, vt",
+				Usage: "visor template file",
+				Value: "visor.template",
+			},
+			cli.StringFlag{
+				Name:  "config-dir, cd",
+				Usage: "config directory path",
+				Value: "./",
+			},
+			cli.StringFlag{
+				Name:  "config-file, cf",
+				Usage: "config file path",
+				Value: "fiber.toml",
+			},
+		},
+		Action: func(c *cli.Context) error {
+			// -- parse flags -- //
+
+			templateDir := c.String("template-dir")
+			templateDir, err := homedir.Expand(templateDir) // expand ~
+			if err != nil {
+				return err
+			}
+
+			coinTemplateFile := c.String("coin-template-file")
+			visorTemplateFile := c.String("visor-template-file")
+
+			// check that the coin template file exists
+			if _, err := os.Stat(filepath.Join(templateDir, coinTemplateFile)); os.IsNotExist(err) {
+				return err
+			}
+			// check that the visor template file exists
+			if _, err := os.Stat(filepath.Join(templateDir, visorTemplateFile)); os.IsNotExist(err) {
+				return err
+			}
+
+			configFile := c.String("config-file")
+			configDir := c.String("config-dir")
+
+			configDir, err = homedir.Expand(configDir)
+			if err != nil {
+				return err
+			}
+
+			configFilepath := filepath.Join(configDir, configFile)
+			// check that the config file exists
+			if _, err := os.Stat(configFilepath); os.IsNotExist(err) {
+				return err
+			}
+
+			coinName := c.String("coin")
+
+			// -- parse template and create new coin.go and visor parameters.go -- //
+
+			config, err := skycoin.NewParameters(configFile, configDir)
+			if err != nil {
+				log.Errorf("failed to create new fiber coin config")
+				return err
+			}
+
+			coinDir := fmt.Sprintf("./cmd/%s", coinName)
+			if _, err := os.Stat(coinDir); os.IsNotExist(err) {
```

Comment:
The err from `os.Stat` needs to be returned if `!os.IsNotExist(err)`

Reply:
looks correct to me. why would I negate an already negative condition?